engine/src/main/java/org/hibernate/validator/internal/util/privilegedactions/SetClassLoader.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.internal.util.privilegedactions;
+
+import java.security.PrivilegedAction;
+
+import org.hibernate.validator.internal.util.Contracts;
+
+/**
+ * Privileged action used to set the Thread context class loader.
+ *
+ * @author Guillaume Smet
+ */
+public final class SetClassLoader implements PrivilegedAction<Void> {
+	private final ClassLoader classLoader;
+
+	public static SetClassLoader ofContext(ClassLoader classLoader) {
```

Comment:
We named these methods `action` in the case where there are not several ones with different semantics.